x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -61,6 +61,12 @@ export class EnterpriseSearchRequestHandler {
     this.enterpriseSearchUrl = config.host as string;
   }
 
+  getBodyAsString(request: KibanaRequest<unknown, unknown, unknown>): string | undefined {
+    if (Buffer.isBuffer(request.body)) return request.body.toString();
+    if (this.isEmptyObj(request.body as object)) return undefined;
+    return JSON.stringify(request.body);
+  }
```

Comment:
This blob only appears to handle `request.body` and no other portions of the request - can we simplify this to passing `this.getBodyAsString(request.body)` instead of the entire request? Does that complicate this too much typing wise?

Reply:
I had it that way originally but changed it because the I couldn't figure out the typing 😬 . I can give it another shot though.

Reply:
no worries!! let me see if I can help out here 🤔 happy to drop this comment if it takes more than 5 mins to figure out lol

Reply:
It looks like you can do this, let me know if you can repro!

```js
  createRequest() {
    this.getBodyAsString(request.body as object);
  }

  getBodyAsString(body: object | Buffer): string | undefined {
    if (Buffer.isBuffer(body)) return body.toString();
    if (this.isEmptyObj(body)) return undefined;
    return JSON.stringify(body);
  }
```

Reply:
Yes, ty! 8fa74b89d11e1d8139b8d18f91ea8c0760c54066